packages/filebrowser-share-file/src/index.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+
+import {
+  Clipboard
+} from '@jupyterlab/apputils';
+
+import {
+  PageConfig,
+  URLExt
+} from '@jupyterlab/coreutils';
+
+import {
+  IFileBrowserFactory
+} from '@jupyterlab/filebrowser';
+
+import {
+  toArray
+} from '@lumino/algorithm';
+
+const shareFile: JupyterFrontEndPlugin<void> = {
+  activate: activateShareFile,
+  id: '@tubitv/filebrowser-extension:share-file',
+  requires: [IFileBrowserFactory],
+  autoStart: true
+};
+
+export default shareFile;
+
+/**
+ * Log debugging information
+ */
+export function log(...args: any[]): void {
+  if (console && console.debug) {
+    console.debug('[filebrowser-share-file]', ...args);
+  }
+}
+
+/**
+ * transform tree url to `/user-redirect/` url for JupyterHub
+ * see https://jupyterhub.readthedocs.io/en/stable/reference/urls.html#user-redirect
+ * @param url
+ */
+export function transformToUserRedirectUrl(url: string): string {
+  return url.replace(/\/user\/([^\/]+)\//, "/user-redirect/");
```

Comment:
what do you mean by saying `Copy shareable link provided by jupyterlab 3.x on Jupyterhub 1.0 does not contain user-redirect.`? jupyterlab 3.x doesn't contain `user-redirect` so we need to utilize this plugin to transform the url?

Reply:
Jupyter Lab 1.x had an issue where the Copy Shareable Link button did not include user-redirect, so different users of JupyterHub could not share links. This is why the tubitv version of this extension was introduced.

This extension was removed from our repository when some outside contributor upgraded to Jupyter Lab 2.x, as the Jupyter Lab folks supposedly addressed this issue.

But in staging where we are running Jupyter Lab 3.x on Jupyter Hub 1.0, the Copy Shareable Link button doesn't provide user-redirect, and so we still need a custom extension override it to substitute user-redirect.